### PR TITLE
Ar 40 class drag selection validation

### PR DESCRIPTION
**Pull request by Tyler-Pamperin**
Changed how prerequisites work to allow for dragging courses, even if they break prerequisites.

Also created a color hierarchy for errors where:
1.) Prerequisite errors are most important and will supersede other errors. This will show up as red.
2.) Fall vs Spring errors are next-most important and will supersede only duplicate course errors. This will show up as orange.
3.) Duplicate course errors are least important and will show up as yellow.

Also made prerequisite error messages more descriptive.

### CODE REVIEW

**Comment by ruudkas** [comment]
Can I make the dumb request of switching the color scheme so yellow is least severe and red is most severe?

I'd be concerned if that was a difficult change but if it is, don't worry about it.

**Comment by ChrisVasterling** [comment]
A couple notes,

Are you aware of this behavior?
image
Taking a class that doesn't have ALL prerequisites satisfied (only direct prerequisites). It's probably fine, this is mostly for documentation. 

Also, when you merged in main, you seem to have removed my changes and broke the completed courses dropdown. 
image
image

**Comment by ruudkas** [comment]
> A couple notes,
> 
> Are you aware of this behavior? image Taking a class that doesn't have ALL prerequisites satisfied (only direct prerequisites). It's probably fine, this is mostly for documentation.
> 
> Also, when you merged in main, you seem to have removed my changes and broke the completed courses dropdown. image image

I can see courses in the completed course dropdown and I have the checkbox. I have a couple css issues that I'm just now finding but that's not from this branch. I also see the issue with non-direct prereqs.

File: academic-requirements/src/components/Semester.tsx [hunk 1]
```diff
@@ -19,6 +17,7 @@ const style: CSSProperties = {
   fontSize: "1rem",
   lineHeight: "normal",
   float: "left",
+  whiteSpace: "pre",
```
Comment (ChrisVasterling) [inline academic-requirements/src/components/Semester.tsx:20]: This line of CSS seems to causing (or is related to) the horizontal scrollbar appearing when hovering a course over a semester. However, removing it causes issues text wrapping for the semester credit counter.

Comment (ruudkas) [reply academic-requirements/src/components/Semester.tsx:20]: It looks like I got the scroll bar fixed, don't really know why it fixed it but 🤷

**Comment by ChrisVasterling** [comment]
image
This is primarily for documentation, but in this edge case the only problem is the first CS145 coming before CS144. The second CS145 coming after CS144 is fine in terms of prerequisites.

**Review by ChrisVasterling — APPROVED** [review]
Looks good (enough) to me.

**Review by ruudkas — APPROVED** [review]
I think it looks good enough that we can put everything else into a bug in the backlog, 99% functioning is much better than none.

**Pull request merged**